server.js

```diff
@@ -43,11 +43,11 @@ const loggedin = (req, res, next) => {
     // user in not logged in
   } else {
     const url = req.url;
-    // check if the endpoint is disallowed
-    if (url === "/home" || url === "/profile") {
-      res.redirect("/login");
-    } else {
+    // visitors are allowed to these endpoints
+    if (url === "/api/post" || url === "/signup" || url === "/login" || url === "/search" || url.includes("/post/")) {
```

Comment:
`/logout` should also be included since the site will bug out if your current session is invalidated by the server and you need to log in again

Reply:
The server will refuse access which makes sense.
The react app will receive a 401 and should then to redirect.
However if makes sense to include it to prevent confusion in the future.
Fixed.